test/jdk/com/sun/jdi/JdbOptions.java

```diff
@@ -49,63 +57,75 @@ static String propString(String name, String value) {
         return "prop[" + name + "] = >" + value + "<";
     }
 
-    public static void main(String[] args) {
-        System.out.println(OK_MSG);
-        // print all args
-        List<String> vmArgs = ManagementFactory.getRuntimeMXBean().getInputArguments();
-        for (String s: vmArgs) {
-            System.out.println(argString(s));
-        }
-        // print requested sys.props
-        for (String p: args) {
-            System.out.println(propString(p, System.getProperty(p)));
+    /**
+     * 1st argument is a filename to redirect application output,
+     * the rest are names of the properties to dump.
+     */
+    public static void main(String[] args) throws IOException {
+        String outFile = args[0];
+        try (PrintStream out = new PrintStream(outFile, StandardCharsets.UTF_8)) {
+            out.println(OK_MSG);
+            // print all args
+            List<String> vmArgs = ManagementFactory.getRuntimeMXBean().getInputArguments();
+            for (String s : vmArgs) {
+                out.println(argString(s));
+            }
+            // print requested sys.props (skip 1st arg which is output filename)
+            for (int i=1; i < args.length; i++) {
+                String p = args[i];
+                out.println(propString(p, System.getProperty(p)));
+            }
         }
     }
 }
 
 public class JdbOptions {
+    private static final String outFilename = UUID.randomUUID().toString() + ".out";
+    private static final Path outPath = Paths.get(outFilename);
     private static final String targ = JbdOptionsTarg.class.getName();
+    private static final String outFileArg = " " + outFilename;
 
     public static void main(String[] args) throws Exception {
         // the simplest case
         test("-connect",
-                "com.sun.jdi.CommandLineLaunch:vmexec=java,options=-client -XX:+PrintVMOptions,main=" + targ)
+                "com.sun.jdi.CommandLineLaunch:vmexec=java,options=-client -XX:+PrintVMOptions,main=" + targ + outFileArg)
```

Comment:
Not to be too nit picky, but I was assuming that the " " would be here and not in outFileName. That would make it much more obvious how the args are separated (eg we are not using commas here).

Reply:
@plummercj are you ok with the latest version?

Reply:
> @plummercj are you ok with the latest version?

Yes. thank you